### PR TITLE
First pass attempt at Decision logic (dev-gated) plus migration of remoteID -> ID

### PR DESCRIPTION
**Pull request by smartin015**

### CODE REVIEW

File: app/Constants.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import {QuestDetails} from './reducers/QuestTypes'
 
-export const NODE_ENV = (process && process.env && process.env.NODE_ENV) || 'dev';
+export const NODE_ENV = (process && process.env && process.env.NODE_ENV) || 'prod';
```
Comment (toddmedema) [inline app/Constants.tsx]: We should also update the hard coded defaults in webpack - https://github.com/ExpeditionRPG/expedition-app/blob/master/webpack.dist.config.js#L34

File: app/Style.scss [hunk 2]
```diff
@@ -1296,6 +1302,17 @@ span.line {
...
   }
 
+  .multiplayer_ripple {
+    height: '100%';
+    width: '100%';
+    position: 'absolute';
+    top: 0;
+    left: 0;
+    overflow: 'hidden';
+    pointerEvents: 'none';
```
Comment (toddmedema) [inline app/Style.scss]: if this is CSS, these should be pointer-events and z-index, no?

File: app/components/views/Tools.tsx [hunk 3]
```diff
@@ -39,13 +39,13 @@ const Tools = (props: ToolsProps): JSX.Element => {
...
       }
-      {!props.settings.simulator && <Button remoteID="1" id="selectQuestCreator" onClick={() => props.onQuestCreatorSelect()}>
+      {!props.settings.simulator && <Button id="selectQuestCreator" onClick={() => props.onQuestCreatorSelect()}>
         <div className="questButtonWithIcon">
           <div className="title">Quest Creator</div>
           <div className="summary">Write your own quests and share them with the world.</div>
         </div>
       </Button>}
-      {!props.settings.simulator && <Button remoteID="2" onClick={() => props.onPrivateQuestsSelect(props.settings, props.user)}>
+      {!props.settings.simulator && <Button id="2" onClick={() => props.onPrivateQuestsSelect(props.settings, props.user)}>
```
Comment (toddmedema) [inline app/components/views/Tools.tsx]: Can this ID be something more useful like #privateQuests?

File: app/components/views/quest/cardtemplates/combat/Actions.tsx [hunk 4]
```diff
@@ -72,12 +73,47 @@ export function generateCombatTemplate(settings: SettingsType, custom: boolean,
...
 
+interface ToDecisionCardArgs {
+  node?: ParserNode;
+  phase: DecisionPhase;
+  numOutcomes?: number;
+  settings?: SettingsType;
+}
+export const toDecisionCard = remoteify(function toDecisionCard(a: ToDecisionCardArgs, dispatch: Redux.Dispatch<any>, getState: () => AppStateWithHistory): ToDecisionCardArgs {
+  if (!a.node) {
+    a.node = getState().quest.node;
```
Comment (toddmedema) [inline app/components/views/quest/cardtemplates/combat/Actions.tsx]: Isn't this logic comparable to the more succinct `a.node = a.node || getState().quest.node`? (same below with settings)

File: app/components/views/quest/cardtemplates/decision/Actions.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+function generateDecisionTemplate(): DecisionState {
+  return {
+    scenario: EMPTY_SCENARIO,
+    numAttempts: 0,
+    outcomes: [],
+    choice: EMPTY_DECISION,
+  };
+}
+
+export function roundTimeMillis(settings: SettingsType, rp?: MultiplayerState) {
```
Comment (toddmedema) [inline app/components/views/quest/cardtemplates/decision/Actions.tsx]: Also TODO dedupe or clarify how this is different than roundTimeMillis in combat (even just calling it skillTimeMillis and using a different constant would be sufficient)

File: app/components/views/quest/cardtemplates/decision/Types.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
+
```
Comment (toddmedema) [inline app/components/views/quest/cardtemplates/decision/Types.tsx]: Lots spaces?

File: package.json [hunk 7]
```diff
@@ -71,6 +71,7 @@
   "dependencies": {
     "@material-ui/core": "^1.0.0-rc.0",
     "@material-ui/icons": "^1.1.0",
+    "@types/classnames": "^2.2.4",
```
Comment (toddmedema) [inline package.json]: Types for the package added, but not the package?

File: app/Constants.tsx (hunk 1 above)
Comment (smartin015) [reply app/Constants.tsx]: Reverted after deciding it's not worth fixing in this PR.

File: app/Style.scss (hunk 2 above)
Comment (smartin015) [reply app/Style.scss]: Done.

File: app/components/views/Tools.tsx (hunk 3 above)
Comment (smartin015) [reply app/components/views/Tools.tsx]: Done.

File: app/components/views/quest/cardtemplates/combat/Actions.tsx (hunk 4 above)
Comment (smartin015) [reply app/components/views/quest/cardtemplates/combat/Actions.tsx]: It is indeed, but it's also a standard style used in the Combat actions and in other actions files. I think we should migrate these wholesale, later.

File: app/components/views/quest/cardtemplates/decision/Actions.tsx (hunk 5 above)
Comment (smartin015) [reply app/components/views/quest/cardtemplates/decision/Actions.tsx]: Done.

File: app/components/views/quest/cardtemplates/decision/Types.tsx (hunk 6 above)
Comment (smartin015) [reply app/components/views/quest/cardtemplates/decision/Types.tsx]: Done.

File: package.json (hunk 7 above)
Comment (smartin015) [reply package.json]: Added

File: app/Style.scss (hunk 2 above)
Comment (smartin015) [reply app/Style.scss]: Done.

**Pull request merged**